apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -257,14 +256,11 @@ export class ApplicationController {
       throw new BadRequestException('Invalid token')
     }
 
-    const existingApplication = await this.applicationService.findOneById(
+    const existingApplication = await this.applicationAccessService.findOneByIdAndNationalId(
```

Comment:
In the assign endpoint, the application is not yet assigned to the authenticated user, so it should just findById.

Reply:
I made the nationalId optional in the `findOneById` function so it can be used here for the assign endpoint